ArduCopter/ReleaseNotes.txt

```diff
@@ -1,5 +1,88 @@
+    d) Guided mode support DO_CHANGE_SPEED commands
+    e) Payload touchdown detection reliability improved    
+    f) Takeoff detection improved to reduce chance of flip before takeoff if GPS moves
+    g) TKOFF_SLEW_TIME allows slower takeoffs in Auto, Guided, etc
+5) RemoteId and SecureBoot enhancements
+    a) Remote update of secure boot's public keys (also allows remote unlocking of bootloader)
+6) Safety enhancements
+    a) Arming checks of FRAME_CLASS/TYPE made mandatory (even if ARMING_CHECK=0)
+    b) crash_dump.bin file saved to SD Card on startup (includes details re cause of software failures)
+    c) Dead-reckoning for 30sec on loss of GPS (requires wind estimation be enabled)
+    d) Dead-reckoning Lua script (On loss of GPS flies towards home for specified number of seconds)
+    e) Disabling Fence clears any active breaches (e.g. FENCE_TYPE = 0 will clear breaches)
+    f) "GPS Glitch" message clarified to "GPS Glitch or Compass error"
+    g) Pre-arm check that configured AHRS is being used (e.g. checks AHRS_EKF_TYPE not changed since boot)
+    h) Pre-arm check that gimbals are healthy (currently only for Gremsy gimbals, others in future release)
+    i) Pre-arm check that all motors are setup
+    j) Pre-arm check that scripts are running
+    k) Pre-arm check that terrain data loaded if RTL_ALT_TYPE set to Terrain
+    l) Pre-arm messages are correctly prefixed with "PreArm:" (instead of "Arm:")
+    m) RC auxiliary switch option for Arm / Emergency Stop
+    n) RC failsafe made pre-arm check (previously only triggered at arming)
+    o) RC failsafe option (see FS_OPTIONS) to continue in Guided obeyed even if GCS failsafe disabled
+    p) TKOFF_RPM_MIN checks all motors spinning before takeoff
+    q) Vibration compensation disabled in manual modes
+7) Scripting enhancements
+    a) CAN2 port bindings to allow scripts to communicate on 2nd CAN bus
+    b) ESC RPM bindings to allow scripts to report engine RPM
+    c) Gimbal bingings to allow scripts to control gimbal
+    d) Pre-arm check bindings (allows scripts to check if pre-arm checks have passed)
+    e) Restarting scripts supported (may require GCS support to be added)
+    f) Semicolon (:) and period (.) supported (e.g both Logger:write() and Logger.write will work)
+    g) VTOL QuickTune script allows easy/safe tuning of quadcopters and multicopters
+8) Sensor driver enhancements
+    a) Benewake H30 radar support
+    b) BMI270 IMU performance improvements
+    c) Logging pause-able with auxiliary switch.  see RCx_OPTION = 165 (Pause Stream Logging)
+    d) Proximity sensor support for up to 3 sensors
+    e) Precision Landing consumes LANDING_TARGET MAVLink message's PositionX,Y,Z fields
+    f) RichenPower generator maintenance-required messages can be suppressed using GEN_OPTIONS param
+    g) TeraRanger Neo rangefinder support
+    h) UBlox GPS support to provide ellipsoid altitude instead of AMSL (see GPS_DRV_OPTIONS)
```

Comment:
most GPS drivers gained ellipsoid support, not just ublox

Reply:
reworded to cover all GPSs.